starter-code/src/viking.js

```diff
@@ -1,11 +1,74 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+this.health = health;
+this.strength = strength
+}
+
+Soldier.prototype.attack = function(){
+    return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage){
+     this.health = this.health - damage;
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = "Harald";
+}
+
+Viking.prototype = Object.create(Soldier.prototype)
+
+
+Viking.prototype.attack = function(){
+    return this.strength;
+}
+Viking.prototype.receiveDamage = function (damage){
+    this.health = this.health - damage;
+    if (this.health === 0){return this.name + " has died in act of combat"}
+    else {
+        return this.name + " has received "+ damage+ " points of damage"
+    }
+}
+
+Viking.prototype.battleCry = function(){
+    return "Odin Owns You All!"
+}
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+
+}
+Saxon.prototype = Object.create(Soldier.prototype)
+
+Saxon.prototype.attack = function(){
+    return this.strength
+}
+Saxon.prototype.receiveDamage = function (damage){
+    this.health = this.health - damage;
+    if (this.health === 0){return "A Saxon has died in combat"}
+    else {
+        return "A Saxon has received "+ damage+ " points of damage"
+    }
+}
 
 // War
-function War() {}
+function War() {
+   this.vikingArmy=[],
+   this.saxonArmy=[]
+}
+
+War.prototype.addViking = function(Viking){
+    this.vikingArmy.push(Viking);
+}
+
+War.prototype.addSaxon = function(Saxon){
+    this.saxonArmy.push(Saxon);
+}
+
+War.prototype.vikingAttack = function(){
```

Comment:
Una posible solucion:

    this.vikingAttack = function(){
        var randomSaxon = Math.floor(Math.random()*this.saxonArmy.length);
        var randomViking = Math.floor(Math.random()*this.vikingArmy.length);
        this.saxonArmy[randomSaxon].receiveDamage(this.vikingArmy[randomViking].strength);
        if (this.saxonArmy[randomSaxon].health<=0) {
            this.saxonArmy.splice(randomSaxon,1);
            return "A Saxon has died in combat";
        } else {
            return "A Saxon has received " + this.vikingArmy[randomViking].strength + " points of damage";
        }
    };